### PR TITLE
Fix search query builder fulltext search

### PR DESCRIPTION
**Pull request by kehh**
In older versions of Providence it was possible to perform searches such as `_fulltext:foobar` via the API or as a part of search query syntax and in the search query builder.

This restores that ability.

### CODE REVIEW

File: app/lib/Plugins/SearchEngine/SqlSearch2.php [hunk 1]
```diff
@@ -382,10 +382,10 @@ private function _processQueryTerm(int $subject_tablenum, $query) {
 	 	}
 	 	$ap = $field ? $this->_getElementIDForAccessPoint($subject_tablenum, $field) : null;
 	 	$words = [$term->text];
-	 	if($field && !is_array($ap)) {
-	 		array_unshift($words, $field);
-	 		$field = null;
-	 	}
+		if($field && $ap && !is_array($ap)) {
```
Comment (collectiveaccess) [inline app/lib/Plugins/SearchEngine/SqlSearch2.php:385]: _getElementIDForAccessPoint() returns null or an array, so we're effectively removing the ability to treat word1:word2 values as text when word1 is not a valid access point here. This behavior is actually needed, so if we want to support _fulltext as an access point we'll need to explicitly test for it.

Comment (kehh) [reply app/lib/Plugins/SearchEngine/SqlSearch2.php:385]: OK so something like: 
```php
<?php
		if($field && $field !== '_fulltext' && !is_array($ap)) {
			array_unshift($words, $field);
			$field = null;
		}
```

**Comment by collectiveaccess** [comment]
Or add code to treat _fulltext as a valid AP in _getElementIDForAccessPoint()

**Comment by kehh** [comment]
> Or add code to treat _fulltext as a valid AP in _getElementIDForAccessPoint()

OK I've tried to do it in that method but come unstuck with all the different return points. Any pointers as to what I should return as an access point definition?

**Comment by kehh** [comment]
OK @collectiveaccess I've adapted this so that when performing a `_fulltext` search it does not send the prefix with the search query. Queries still parse correctly

**Pull request merged**